smoke_tests/smoke_tests.py

```diff
@@ -0,0 +1,94 @@
+import argparse
+import sys
+import unittest
+from unittest import TestSuite
+import requests
+
+from tests.smoke_test_case import SmokeTestCase
+from tests.unauthenticated.status_tests import StatusTests
+from tests.authenticated.azure.managed_apps_tests import ManagedAppsTests
+from tests.authenticated.billing_profiles import BillingProfileTests
+
+DESCRIPTION = """
+BPM Smoke Test
+Enter the host (domain and optional port) of the BPM instance you want to to test. This test will ensure that the BPM 
+instance running on that host is minimally functional.
+"""
+
+
+def gather_tests(is_authenticated: bool = False, azure_subscription: bool = False) -> TestSuite:
+    suite = unittest.TestSuite()
+    status_tests = unittest.defaultTestLoader.loadTestsFromTestCase(StatusTests)
+    suite.addTests(status_tests)
+    if is_authenticated:
+        profile_tests = unittest.defaultTestLoader.loadTestsFromTestCase(BillingProfileTests)
+        suite.addTests(profile_tests)
+        if azure_subscription:
+            managed_apps_tests = unittest.defaultTestLoader.loadTestsFromTestCase(ManagedAppsTests)
+            suite.addTests(managed_apps_tests)
+    return suite
+
+
+def main(main_args):
+    SmokeTestCase.BPM_HOST = main_args.bpm_host
+    SmokeTestCase.USER_TOKEN = main_args.user_token
+    ManagedAppsTests.AZURE_SUBSCRIPTION_ID = main_args.azure_sub_id
+
+    valid_user_token = main_args.user_token is not None and verify_user_token(main_args.user_token)
```

Comment:
FWIW this can be done programmatically https://github.com/broadinstitute/dsp-scripts/blob/master/azure/workspaces/tools/utils/auth.py#L20